site/content/entry/citra-progress-report-2017-september.md

```diff
@@ -0,0 +1,360 @@
+
+When a program is written in a high-level programming language, such as C++, Rust,
+or Go, before the program can be run on a machine, it must be translated or "compiled"
+to machine code. Although it is possible to do this translation by hand, it is
+usually extremely difficult to do so and very time consuming. So instead, we have
+a program called a compiler than can automatically do this translation for us.
+
+This is also why a program compiled for an ARM machine cannot be run directly on
+an x86 machine, even when the source can work on either machine without issues.
+Instead this program must be translated, interpreted, or recompiled from source
+to x86. (In fact, this translation is exactly what [dynarmic](https://github.com/MerryMage/dynarmic)
+does to run code from a 3DS.)
+
+Every statement in a program must have an exact, unambiguous definition of what
+it does (its semantics). But, in the same way that a statement that means one
+thing can be written many different ways, a different compilers can translate the
+same statement many different ways.
+
+On Windows, there's two popular C++ compilers available as of today: MSVC++, which
+is the compiler Microsoft has written for Windows, and MINGW GCC, which is actually
+a port of the Linux `gcc` compiler to Windows. For better or worse, MINGW GCC
+compiles to a much more efficient binary than MSVC++, and so [jroweboy](https://github.com/jroweboy)
+has changed the Citra AppVeyor build script to use MINGW GCC instead of MSVC++.
+This change also has closed the gap in performance the new Nightly builds had
+compared to the old Bleeding Edge builds.
+
+## [Load different shared font depending on the region](https://github.com/citra-emu/citra/pull/2915) by [wwylele](https://github.com/wwylele)
+
+Remember that last month [wwylele](https://github.com/wwylele) changed Citra so
+that instead of loading the shared font from a seperate file, it would
+[load it from the system archive](https://citra-emu.org/entry/citra-progress-report-2017-august/)?
+This builds on top of that behaviour. You see, a 3DS doesn't have a shared font,
+it has *four*. One contains glyps for Latin script (for English, Spanish, Italian,
+French, etc.) and Japanese scripts, another contains glyps for Traditional Chinese,
+the third font contains the ones for Simplified Chinese, and the last font contains
+the ones for Korean.
+
+Before this PR, Citra would simply load the first shared font regardless of game
+or region. This made non-Latin or non-Japanese script games display completely
+incorrect characters at best, or crash at worst. Now Citra will load the appropriate
+shared font from the system archive depending on the region selected, just like
+a real console! Though, this will not work on machines that only have the
+`shared_font.bin` file, because it only contains the first of the four shared
```

Comment:
Sorry for not mentioning this in the interview. This is not correct. `shared_font.bin` only contains the one **corresponding to the device region**. For example, a Korean 3DS console will dump the Korean font to shared_font.bin, which allows all Korean game work (which are the users supposed to own). In this sense, the first font archive PR actually broke all non-JPN/USA/EUR games which used to work on citra, and this, the second PR, is to restore that compatibility.